cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQueryBuilder.java

```diff
@@ -0,0 +1,613 @@
+                unitId = getUnitIdFromOrganization(parentUnitId);
+            }
+            cristinQuery.setPath(PATH_ORGANISATION, unitId);
+        }
+        return this;
+    }
+
+    /**
+     * Setter a participant of the project by Cristin id, name or part of the name.
+     */
+    public CristinQueryBuilder withParticipant(String participant) {
+        cristinQuery.setValue(PROJECT_PARTICIPANT, participant);
+        return this;
+    }
+
+    /**
+     * Setter Identity.
+     */
+    public CristinQueryBuilder withPathIdentity(String identity) {
+        System.out.println("withPathIdentity -> " + identity);
+        if (nonNull(identity) && !identity.isBlank()) {
+            if (identity.matches(PATH_ORGANISATION.getPattern())) {
+                cristinQuery.setPath(PATH_ORGANISATION, identity);
+            } else if (identity.matches(PATH_PROJECT.getPattern())) {
+                cristinQuery.setPath(PATH_PROJECT, identity);
+            } else {
+                cristinQuery.setPath(IDENTITY, identity);
+            }
+        } else {
+            cristinQuery.setPath(IDENTITY, EMPTY_STRING);
+        }
+        return this;
+    }
+
+    /**
+     * Setter Organization.
+     */
+    public CristinQueryBuilder withPathOrganization(String organization) {
+        System.out.println("withPathOrganization -> " + organization);
+
+        if (nonNull(organization)) {
+            if (organization.matches(PROJECT_ORGANIZATION.getPattern())) {
+                cristinQuery.setValue(PROJECT_ORGANIZATION,organization);
+            } else {
+                cristinQuery.setPath(PATH_ORGANISATION, organization);
+            }
+        } else {
+            cristinQuery.setValue(PATH_ORGANISATION, EMPTY_STRING);
+        }
+        return this;
+    }
+
+    /**
+     * Setter Project(identity) .
+     */
+    public CristinQueryBuilder withPathProject(String project) {
+        System.out.println("withPathProject -> " + project);
+        cristinQuery.setPath(PATH_PROJECT, project);
+        return this;
+    }
+
+
+    /**
+     * Setter project manager by Cristin id, name or part of the name.
+     */
+    public CristinQueryBuilder withProjectManager(String projectManager) {
+        cristinQuery.setValue(PROJECT_MANAGER, projectManager);
+        return this;
+    }
+
+    /**
+     * Setter Query.
+     */
+    public CristinQueryBuilder withQuery(String query) {
+        if (nonNull(query)) {
+            var key = grantOrTitleKey(query);
+            cristinQuery.setValue(key, query);
+        } else {
+            cristinQuery.setValue(QUERY, EMPTY_STRING);
+        }
+        return this;
+    }
+
+    /**
+     * Setter status of projects.
+     */
+    public CristinQueryBuilder withStatus(String status) {
+        var statusKind = attempt(() ->
+            ProjectStatus.valueOf(status.toUpperCase(Locale.getDefault())).name())
+            .orElse((e) -> EMPTY_STRING);
+        cristinQuery.setValue(STATUS, statusKind);
```

Comment:
will status always be appended to parameters even if not sent?

Reply:
nope, 
`
case STATUS:
        withStatus(value);
        break;
`